redwood-treehouse-host/src/iosMain/kotlin/app/cash/redwood/treehouse/TreehouseUIView.kt

```diff
@@ -60,13 +67,73 @@ public class TreehouseUIView private constructor(
   private class RootUiView : UIView(cValue { CGRectZero }) {
     lateinit var treehouseView: TreehouseUIView
 
+    private var sizeThatFits = CGSizeMake(0.0, 0.0)
+
+    /**
+     * We've got a few things making layouts difficult:
+     *
+     *  * This view may be itself used with either Auto Layout or manual layout (like [sizeThatFits]
+     *    and [setFrame]).
+     *  * The contents are updated dynamically.
+     *
+     * It seems the simplest thing to do is to override [layoutSubviews], and to call
+     * [invalidateIntrinsicContentSize] from within that if the frame we're given isn't the size
+     * we want.
+     *
+     * https://mischa-hildebrand.de/en/2017/11/the-auto-layout-comprehendium/
+     * https://forums.developer.apple.com/forums/thread/682973
+     */
     override fun layoutSubviews() {
+      super.layoutSubviews()
+
+      val oldWidthThatFits = sizeThatFits.useContents { width }
+      val oldHeightThatFits = sizeThatFits.useContents { height }
+
+      // TODO(jessewilson): is passing the screen bounds to sizeThatFits() appropriate?
+      this.sizeThatFits = sizeThatFits(
+        UIScreen.mainScreen.bounds.useContents<CGRect, CValue<CGSize>> {
```

Comment:
Seems questionable. Root views are not guaranteed to occupy a full screen. Presumably they could also render on other screens.

Reply:
Yep, agreed.